### PR TITLE
schedulers: extract balancePlan for balancers

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve?

Fix https://github.com/tikv/pd/issues/835

### What is changed and how it works?

Extract `balancePlan` so we don't have to pass all parameters around.

### Check List

Tests
- Unit test

### Release note
```release-note
None
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: server/schedulers/utils.go [hunk 1]
```diff
@@ -36,61 +36,102 @@ const (
...
+
+func (p *balancePlan) GetOpInfluence(storeID uint64) int64 {
+	return p.opInfluence.GetStoreInfluence(storeID).ResourceProperty(p.kind)
+}
+
+func (p *balancePlan) SourceStoreID() uint64 {
+	return p.source.GetID()
+}
+
+func (p *balancePlan) SourceStoreLabel() string {
```
Comment (rleungx) [inline server/schedulers/utils.go:68]: The function is a little bit misleading since we have a store label actually but with a different meaning.

Comment (disksing) [reply server/schedulers/utils.go:68]: good point.

**Review by rleungx — APPROVED** [review]

**Pull request merged**